dmapiclient/base.py

```diff
@@ -91,6 +88,17 @@ def _request(self, method, url, data=None, params=None):
         base_url = urlparse.urlparse(self.base_url)
         url = url._replace(netloc=base_url.netloc, scheme=base_url.scheme).geturl()
 
+        r = requests.PreparedRequest()
+        r.prepare_url(url=url, params=params)
```

Comment:
A direct request doesn't give us access to the full URL without sending the request. For Direct Award projects we need to generate the full Search API URL to store it in our database but don't need to actually send the request, so we are using the request lib's intermediate step (PreparedRequest) to build the URL and expose it to the other part of our application as required.